path_role.go

```diff
@@ -304,13 +311,22 @@ func (b *kubeAuthBackend) pathRoleCreateUpdate(ctx context.Context, req *logical
 
 	if namespaces, ok := data.GetOk("bound_service_account_namespaces"); ok {
 		role.ServiceAccountNamespaces = namespaces.([]string)
-	} else if req.Operation == logical.CreateOperation {
-		role.ServiceAccountNamespaces = data.Get("bound_service_account_namespaces").([]string)
 	}
-	// Verify namespaces is not empty
-	if len(role.ServiceAccountNamespaces) == 0 {
-		return logical.ErrorResponse("%q can not be empty", "bound_service_account_namespaces"), nil
+
+	role.ServiceAccountNamespaceSelector = data.Get("bound_service_account_namespace_selector").(string)
+
+	// Verify namespaces is not empty unless selector is set
+	if len(role.ServiceAccountNamespaces) == 0 && role.ServiceAccountNamespaceSelector == "" {
```

Comment:
nit: I see that we take the length of `role.ServiceAccountNamespaces` in multiple places, would be nice to factor that out to variable.